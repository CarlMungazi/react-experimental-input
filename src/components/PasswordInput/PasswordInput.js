import React from 'react';
import ProgressBar from '../ProgressBar';
import Icon from '../Icon';
import Input from '../Input';

class PasswordInput extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      showPassword: false
    }
  }

  toggleShowPassword = evt => {
    this.setState(prevState => {
      return { showPassword: !prevState.showPassword };
    });

    if (evt) evt.preventDefault();
  }

  render() {
    const { value, minLength, onChange, placeholder, required, showVisibilityToggle, quality, error } = this.props;
    const { showPassword } = this.state;

    return (
      <Input 
        value={value}
        minLength={minLength}
        onChange={onChange}
        placeholder={placeholder}
        required={required}
        error={error}
        type={showPassword ? "text" : "password"}
      >
        {
          showVisibilityToggle &&
          <button href="" onClick={this.toggleShowPassword} style={{ marginLeft: 5 }}> <Icon /> </button>
        }
        {
          value.length > 0 && quality && <ProgressBar percent={quality} width={130} />
        }
      </Input>
    )

  }
}

export default PasswordInput;