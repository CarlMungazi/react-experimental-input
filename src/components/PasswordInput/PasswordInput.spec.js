import React from 'react';
import renderer from 'react-test-renderer';
import PasswordInput from './PasswordInput';
import {shallow} from 'enzyme';

test('toggle input type when show/hide password is clicked', () => {
  const wrapper = shallow(
    <PasswordInput 
      onChange={() => {}}
      value=""
      showVisibilityToggle
    />
  );

  // Password input should have a type of password set initially
  expect(wrapper.find({type: "password"})).toHaveLength(1);
  expect(wrapper.find({type: "text"})).toHaveLength(0);

  // Password input should have a type of text after clicking toggle
  wrapper.find('a').simulate('click');
  expect(wrapper.find({type: "password"})).toHaveLength(0);
  expect(wrapper.find({type: "text"})).toHaveLength(1);
})

test('hides password quality by default', () => {
  const tree = renderer.create(
    <PasswordInput 
      onChange={() => {}}
      value="Uisi38#8iad"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
})

test('shows password quality when enabled and a password is entered', () => {
  const tree = renderer.create(
    <PasswordInput 
      onChange={() => {}}
      value="Uisi38#8iad"
      showQuality
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
})

test('hides password quality when enabled but no password is entered', () => {
  const tree = renderer.create(
    <PasswordInput 
      onChange={() => {}}
      value=""
      showQuality
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
})