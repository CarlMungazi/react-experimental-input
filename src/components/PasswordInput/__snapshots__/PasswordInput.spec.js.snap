// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality by default 1`] = `
<div>
  <input
    minLength=""
    onChange={[Function]}
    placeholder="Enter your text here"
    required={false}
    type="password"
    value="Uisi38#8iad"
  />
</div>
`;

exports[`hides password quality when enabled but no password is entered 1`] = `
<div>
  <input
    minLength=""
    onChange={[Function]}
    placeholder="Enter your text here"
    required={false}
    type="password"
    value=""
  />
</div>
`;

exports[`shows password quality when enabled and a password is entered 1`] = `
<div>
  <input
    minLength=""
    onChange={[Function]}
    placeholder="Enter your text here"
    required={false}
    type="password"
    value="Uisi38#8iad"
  />
</div>
`;
